src/Chullo.php

```diff
@@ -58,56 +62,49 @@ public function getBaseUri() {
      * @param array     $headers        HTTP Headers
      * @param string    $transaction    Transaction id
      *
-     * @return mixed    String or binary content if 200. Null if 304.
+     * @return mixed    Full response if found.  Null otherwise.
      */
-    public function getResource($uri = "",
-                                $headers = [],
-                                $transaction = "") {
-        // Set headers
-        $options = ['headers' => $headers];
-
-        // Ensure uri takes transaction into account.
-        $uri = $this->prepareUri($uri, $transaction);
-
-        // Send the request.
-        $response = $this->client->request(
-            'GET',
+    public function getResource(
+        $uri = "",
+        $headers = [],
+        $transaction = ""
+    ) {
+        $response = $this->api->getResource(
             $uri,
-            $options
+            $headers,
+            $transaction
         );
-
-        $code = $response->getStatusCode();
-
-        if ($code == 304) {
+        if ($response->getStatusCode() != 200) {
             return null;
         }
 
-        return $response->getBody();
+        return $response;
```

Comment:
You are probably right (100% right), but maybe it could be also an option to add a specific method for getting only the body() instead of having the same method name returning different things (like a getResourceContent in the FedoraClient Interface?). Just an idea, i'm sure you have good enough reasons. It's just i like the idea of obscuring a few errors when returning for the services but preserve headers, etc in case of success. Just an idea :smile: , since we really only need the clean body for https://github.com/Islandora-CLAW/chullo/pull/30/files#diff-9a4a06deeec43205acd722cefedf7f05R141